Chrome: Try reorganizing the editor's header

youknowriad opened the pull request:
## Description
This changes the UI of the Editor's moving all the content related actions to the left and the document related actions to the right.

## Screenshots (jpeg or gifs if applicable):
<img width="943" alt="screen shot 2017-10-04 at 16 56 35" src="https://user-images.githubusercontent.com/272444/31185880-084820f0-a925-11e7-8aa8-87f57a3a90ca.png">

youknowriad committed:
Chrome: Reorganize the editor's header

youknowriad committed:
Chrome: Reorganize the editor's header

mtias commented:
I'm liking this a lot.

mtias commented on editor/header/index.js:
```diff
@@ -59,22 +71,52 @@ function Header( { multiSelectedBlockUids, onRemove, onDeselect } ) {
-			<ModeSwitcher />
-			<SavedState />
-			<Tools />
+			<div className="editor-header__left">
```
Can we use a more semantic class? Maybe `editor-header__content-tools` and `editor-header__settings` for the right one?

mtias commented on editor/header/index.js:
```diff
@@ -59,22 +71,52 @@ function Header( { multiSelectedBlockUids, onRemove, onDeselect } ) {
+			<div className="editor-header__left">
+				<Inserter position="bottom right" />
+				<IconButton
+					className="editor-tools__undo"
```
Shouldn't the namespace be the file component?

youknowriad replied:
yep sorry, copy/paste

mtias commented on editor/header/publish-button/style.scss:
```diff
@@ -0,0 +1,35 @@
+.editor-publish-button.is-saving:disabled {
+	position: relative;
+
+	// These styles overrides the disabled state with the button primary styles
```
@jasmussen is this something we need upstream?

jasmussen replied:
Good question. I don't know what the best approach is, here — looking at it now I'm worried that we might have a blue disabled button, even if a user has a different admin color scheme.

Long term one would hope we had a single "button-primary" component that has this all baked in, instead of multiple disparate CSS that's competing.

Commit:
Polish the header a bit

- Normalize the margins between items
- Tweak the alignment and size of the publish button
- Make the Settings cog have a background when toggled, and match the size of the publish button

jasmussen commented:
A lot of this was inspired by @afercia and @hedgefield and #467, particularly the discussions around grouping the save state indicator with the publish button. I'm curious of your thoughts on this design, Andrea.

afercia commented:
@jasmussen grouping and separating content related actions and document/editor related actions makes lot of sense to me. It is a nice improvement.

Also the logical sequence:
`Is everything saved? Yes > Let me check how it looks. Preview > OK Publish`

seems good to me.

Not sure about a couple details and it doesn't address the sidebar toggle issue yet but to me this is an improvement. 🙂  I've probably missed some of the recent changes, quick questions:
- what about the visual/text mode switcher?
- what the ellipsis button is for?

jasmussen commented:
> what about the visual/text mode switcher?
> what the ellipsis button is for?

I think this got buried in another ticket or PR, my apologies, but it is related to the PR to allow users to edit the HTML on a per-block basis (#2797). It's not fully fleshed out yet, but basically the ellipsis would _be_ the mode switch:

<img width="430" alt="screen shot 2017-10-05 at 11 28 41" src="https://user-images.githubusercontent.com/1204802/31220425-b21a0ca6-a9c0-11e7-8c27-0c271e0c1340.png">

It could potentially also be pluggable, so global actions like for example spell check, could live there.

afercia commented:
Ah yes the ellipsis thing is also on https://github.com/WordPress/gutenberg/issues/2460 
Reading all the considerations about mobile in #2797 separating the two group of controls makes even more sense to me.
